### PR TITLE
Shiny Express support

### PR DESCRIPTION
**Pull request by jcheng5**
## Testing notes

1. Create a virtual environment somewhere on the server (e.g. `/srv/shiny-server/.venv`) and install shiny into it
2. In /etc/shiny-server/shiny-server.conf, add the directive `python /srv/shiny-server/.venv;`
3. Create /srv/shiny-server/express-app/app.py, with the following contents:

```python
from shiny.express import input, render, ui

ui.input_slider("n", "N", 0, 100, 20)

@render.text
def txt():
    return f"n*2 is {input.n() * 2}"
```

4. Start (or restart) shiny-server and load http://localhost:3838/express-app/

### CODE REVIEW

File: python/SockJSAdapter.py [hunk 1]
```diff
@@ -209,12 +211,21 @@ def run():
         os.environ["RSTUDIO_PANDOC"] = input["pandocPath"]
 
     sys.path.insert(0, input["appDir"])
-    app_module = importlib.import_module("app")
-    app = getattr(app_module, "app")
+
+    app_file = os.path.join(input["appDir"], "app.py")
+    if is_express_app(app_file, None):
+        print("Shiny Express detected")
```
Comment (wch) [inline python/SockJSAdapter.py:217]: Just curious: is this `print` statement and the one below intentional, or leftover debugging info?

**Review by wch — APPROVED** [review]

File: python/SockJSAdapter.py (hunk 1 above)
Comment (jcheng5) [reply python/SockJSAdapter.py:217]: I had intended for it to be intentional. But stdout is actually thrown away by Shiny Server, so right now it's useless. I should remove it or have it print to stderr. (Probably remove it, I guess?)

Comment (jcheng5) [reply python/SockJSAdapter.py:217]: Also debating whether to have stdout redirect to stderr (and thus the error log) in Shiny for Python apps. There's reasons why we don't do this for R, but I think it is more defensible in Python, especially given that some of the framework's log messages are currently on stdout 😬

**Comment by jcheng5** [comment]
I had to add a couple of commits to get stdout->stderr redirection working right--a little more subtle than it seems. I wonder if Connect and Posit Cloud are doing anything with stdout. We really should get on fixing py-shiny so it doesn't emit errors to stdout ever!

**Comment by wch** [comment]
Looks good to me. The stdout/stderr stuff seems like a big hammer, though. Should we just fix py-shiny instead, so that errors always go to stderr? Then we wouldn't have to worry about what Connect or Cloud do with stdout.

**Pull request merged**